### ISSUE TITLE
compile-time variable assignments through #[cfg] configuration cause incorrect 'variable unused' error

### ISSUE THREAD

**Issue by AbirV** [intellij-rust/intellij-rust#7892]
IntelliJ Rust plugin version: 0.4.155.4114-212
Rust toolchain version: Visual Studio 2019 Community - 16.0. CMake - Bundled: 3.20.2. Debugger - Bundled LLDB: 9.0
IDE name and version: CLion 2021.2.2 (#CL-212.5284.51)
Operating system: Windows 10 Pro 21H1 (19043.1237)

Problem description: 
An incorrect warning ("Variable '<name>' is never used") is displayed when using compiler #[cfg] flags for build-specific variable assignment. Syntax highlighting shows that those statements are not going to compile on the current environment, but linting is apparently still done on both statements. 
The warning only appears on the first let statement, but after compilation only one such statement would actually exist. 

Steps to reproduce:

```
#[cfg(windows)]
let line_terminator = "\r\n";
#[cfg(linux)]
let line_terminator = "\n";

println!(
    f,
    "Hello {line_term}\t World!{line_term}"
    line_term = line_terminator
)
```

Although the error also occurs with other #[cfg] flags - 

```
#[cfg(debug_assertions)]
let debug_flag = true;
#[cfg(not(debug_assertions))]
let debug_flag = false;

if debug_flag {
<etc>
```

**Comment by vlad20012** [comment]
The first example is not reproducible. It (not) works only with `debug_assertions`

**Comment by AbirV** [comment]
![image](https://user-images.githubusercontent.com/46656642/134761054-a57bafd9-8d52-4fd2-b9af-3448e981ed82.png)

**Comment by AbirV** [comment]
![image](https://user-images.githubusercontent.com/46656642/134761093-d588131d-5662-41ca-ac92-88869afc59b5.png)

**Comment by vlad20012** [comment]
Thanks for the screenshot!
I was not able to reproduce it because I'm on Linux.

**Comment by AbirV** [comment]
Oh, interesting. Thanks for the input.

**Comment by AbirV** [comment]
What if you reverse the direction of the statements, putting the active statement first?

**Comment by vlad20012** [comment]
Yep, it reproduces now. It looks like we ignore `cfg` attributes on let statements during name resolution.
